### PR TITLE
Implement ServerMock#permissionMessage and getPermissionMessage

### LINKED ISSUE #595: Support for ServerMock#permissionMessage()
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Describe the feature

MockBukkit 2.0 switched to Paper which introduced `ServerMock#getPermissionMessage` and `ServerMock#permissionMessage`, which are currently unimplemented.

### PR DESCRIPTION
**Pull request by Benlewis9000**
# Description
Implement ServerMock#permissionString and ServerMock#getPermissionString.
Code was not compiling without adding unimplemented methods to BlockMock and BlockDataMock, so they have been included.

# Checklist
The following items should be checked before the pull request can be merged.
- [x] Unit tests added.
- [x] Code follows existing code format.

# Info on creating a pull request
- Make sure that unit tests are added which test the relevant changes.
- Make sure that the changes follow the existing code format.

# For maintainer
When a PR is approved, the maintainer should label this PR with `release/*`.

- If the PR fixes a bug in MockBukkit, update the patch version. (if the current version is `0.4.1`, the new version should be `0.4.2`)
- If the PR adds a new feature to MockBukkit, update minor version. (if the current version is `0.4.1`, the new version should be `0.5.0`)

Note that a PR that fixes an `UnimplementedOperationException` should be considered a new version and not a bugfix.

### CODE REVIEW

File: src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java [hunk 1]
```diff
@@ -457,6 +457,20 @@ public boolean breakNaturally(@NotNull ItemStack tool, boolean triggerEffect)
...
+		throw new UnimplementedOperationException();
+	}
+
+	@Override
+	public void randomTick()
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java]: Those Methods seem to be out of scope for this PR, feel free to open a separate PR to fix those

File: src/main/java/be/seeseemelk/mockbukkit/block/data/BlockDataMock.java [hunk 2]
```diff
@@ -139,6 +139,13 @@ public boolean isFaceSturdy(@NotNull BlockFace face, @NotNull BlockSupport suppo
 
 	}
 
+	@Override
+	public boolean isRandomlyTicked()
+	{
+		// TODO Auto-generated method stub
+		throw new UnimplementedOperationException();
+	}
+
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/block/data/BlockDataMock.java]: Those Methods seem to be out of scope for this PR, feel free to open a separate PR to fix those

**Review by thelooter — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java [hunk 3]
```diff
@@ -1934,18 +1935,18 @@ public boolean suggestPlayerNamesWhenNullTabCompletions()
...
 	}
 
+	private static final Component noPermission = Component.text("I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.", NamedTextColor.RED);
+
 	@Override
 	public @NotNull String getPermissionMessage()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return net.kyori.adventure.text.serializer.legacy.LegacyComponentSerializer.legacyAmpersand().serialize(noPermission);
```
Comment (thelooter) [inline src/main/java/be/seeseemelk/mockbukkit/ServerMock.java]: You can use the Instance from MockUnsafeValues

**Review by thelooter — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java [hunk 4]
```diff
@@ -1934,18 +1935,18 @@ public boolean suggestPlayerNamesWhenNullTabCompletions()
 		throw new UnimplementedOperationException();
 	}
 
+	private static final Component noPermission = Component.text("I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.", NamedTextColor.RED);
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/ServerMock.java]: Constants should be upper snake_case (`NO_PERMISSION`) and defined at the top of the class

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java (hunk 3 above)
Comment (Insprill) [reply src/main/java/be/seeseemelk/mockbukkit/ServerMock.java]: or at least use an import :P

**Review by Insprill — CHANGES_REQUESTED** [review]

File: src/main/java/be/seeseemelk/mockbukkit/ServerMock.java [hunk 5]
```diff
@@ -142,6 +143,8 @@ public class ServerMock extends Server.Spigot implements Server
 {
 
 	private static final Component MOTD = Component.text("A Minecraft Server");
+	private static final Component NO_PERMISSION = Component.text("I'm sorry, but you do not have permission to perform this command. Please contact the server administrators if you believe that this is in error.", NamedTextColor.RED);
+
```
Comment (Insprill) [inline src/main/java/be/seeseemelk/mockbukkit/ServerMock.java:147]:
```suggestion

**Review by Insprill — CHANGES_REQUESTED** [review]

**Review by Insprill — APPROVED** [review]
LGTM

**Review by thelooter — APPROVED** [review]

**Pull request merged**